sway-core/src/asm_generation/fuel/fuel_asm_builder.rs

```diff
@@ -1627,18 +1654,42 @@ impl<'ir, 'eng> FuelAsmBuilder<'ir, 'eng> {
                 ))
             }
         } else {
+            let stored_ty = stored_val.get_type(self.context).unwrap();
+            let byte_len = ir_type_size_in_bytes(self.context, &stored_ty);
+
             let dst_reg = self.value_to_register(dst_val)?;
             let val_reg = self.value_to_register(stored_val)?;
 
-            self.cur_bytecode.push(Op {
-                opcode: Either::Left(VirtualOp::SW(
-                    dst_reg,
-                    val_reg,
-                    VirtualImmediate12 { value: 0 },
-                )),
-                comment: "store value".into(),
-                owning_span,
-            });
+            match byte_len {
```

Comment:
Same as above.

Reply:
Bigger than 8 is also fine. But I simplified a little bit nonetheless..